File: fixtures/scale/haskell/large/Core/Registry.hs
module Core.Registry (Entry(..), catalog, size) where

import qualified Ciphers.Aes128.Runner as Aes128
import qualified Ciphers.Blowfish256.Runner as Blowfish
import qualified Ciphers.Chacha20256.Runner as Chacha20
import qualified Ciphers.Salsa20256.Runner as Salsa20

data Entry = Entry { entryLabel :: String }

catalog :: [Entry]
catalog =
  [ Entry Aes128.label
  , Entry Chacha20.label
  , Entry Salsa20.label
  , Entry Blowfish.label
  ]

size :: Int
size = length catalog

File: fixtures/scale/elixir/large/lib/core/registry.ex
defmodule Core.Registry do
  alias Ciphers.Aes128.Runner, as: Aes128
  alias Ciphers.Blowfish256.Runner, as: Blowfish
  alias Ciphers.Chacha20256.Runner, as: Chacha20
  alias Ciphers.Salsa20256.Runner, as: Salsa20

  def catalog do
    [Aes128.label(), Chacha20.label(), Salsa20.label(), Blowfish.label()]
  end

  def size, do: length(catalog())
end
